test/integration/scripts.test.js

```diff
@@ -0,0 +1,52 @@
+const expect = require('expect');
+const fs = require('fs');
+const path = require('path');
+const { spawnSync } = require('child_process');
+
+const rootDir = path.resolve(__dirname, '..', '..');
+const dataDir = path.resolve(rootDir, 'test', 'integration', 'data');
+
+describe("Command-Line Scripts", () => {
+  fs.readdirSync(path.resolve(rootDir, 'test', 'integration', 'data', 'source')).forEach(example => {
+    describe(example, () => {
+      const sourcePath = path.resolve(dataDir, 'source', example)
+      const redactedPath = path.resolve(dataDir, 'redacted', example)
+      const translatedPath = path.resolve(dataDir, 'translated', example)
+      const restoredPath = path.resolve(dataDir, 'restored', example)
```

Comment:
The one thing that occurs to me, is I would have organized the test files by test case rather than by role.  That is, you've got:
```
data/
  source/
    ordering.md
    questions.md
  redacted/
    ordering.md
    questions.md
  restored/
    ordering.md
  translated/
    ordering.md
```

I think I would have instead:
```
data/
  ordering/
    source.md
    redacted.md
    restored.md
    translated.md
  questions/
    source.md
    redacted.md
```

I can see pros and cons to each approach.  It might be really annoying to have a dozen `source.md` files in the repo and differentiate by directory name, but grouping them by case would ensure that related fixtures show up together in GitHub diffs, and might it more obvious to a new dev how a new test case should be added.

I think it's okay either way, just wanted to suggest it.

Reply:
Oh, that's actually a huge improvement. I was wanting to also add some tests around switching between formats, and wasn't sure how to do that with the existing pattern, but this organization will make that much more straightforward.